protobuf/key_attributes.proto

```diff
@@ -18,62 +18,59 @@ syntax = "proto3";
 
 package key_attributes;
 
+message KeyAttributesProto {
+  KeyType key_type = 1;
+  oneof algorithm {
+    Cipher cipher = 100;
+    AsymmetricEncryption asymmetric_encryption = 101;
+    MAC mac = 102;
+    AEAD aead = 103;
+    Sign sign = 104;
+    KeyDerivation key_derivation = 105;
+    KeyAgreement key_agreement = 106;
+    Hash hash = 107;
+  }
+  uint32 key_size = 3;
+  UsageFlags flags = 4;
+}
+
 enum KeyType {
-  HMAC_Key = 0;
-  Derivation_Key = 1;
-  AES_Key = 2;
-  DES_Key = 3;
-  Camellia_Key = 4;
-  ARC4_Key = 5;
-  RSA_Public_Key = 6;
-  RSA_Keypair = 7;
-  DSA_Public_Key = 8;
-  DSA_Keypair = 9;
+  Raw_Data = 0;
+  HMAC_Key = 1;
```

Comment:
I think there was a reason for it at some point (protobuf not supporting an enum variant and something else being named the same, in one file?), but I think it can be removed now

Reply:
Found out why. It's not possible to have the same variant name twice in the same file